drakrun/drakrun/analyzer.py

```diff
@@ -0,0 +1,515 @@
+            except subprocess.TimeoutExpired as e:
+                log.error("DRAKVUF timeout expired")
+                raise e
+
+        if options.raw_memory_dump:
+            vm.vm.memory_dump(str(output_dir / "post_sample.raw_memdump.gz"))
+
+    log.info("Analysis done. Collecting artifacts...")
+
+    # Make sure dumps have a reasonable size.
+    # Calculate dumps_metadata as it's required by the `analysis` task format.
+    dumps_path = output_dir / "dumps"
+    dumps_zip_path = output_dir / "dumps.zip"
+    dumps_metadata = crop_dumps(str(dumps_path), str(dumps_zip_path))
+
+    # Compress IPT traces, they're quite large however they compress well
+    ipt_path = output_dir / "ipt"
+    ipt_zip_path = output_dir / "ipt.zip"
+    compress_ipt(str(ipt_path), str(ipt_zip_path))
+
+    time_finished = time.time()
+    return {
+        "time_started": time_started,
+        "time_finished": time_finished,
+        "start_command": start_command,
+        "dumps_metadata": dumps_metadata,
+    }
+
+
+class PluginsArgAction(argparse.Action):
+    def __init__(self, option_strings, dest, nargs=None, **kwargs):
+        if nargs is not None:
+            raise ValueError("nargs not allowed")
+        super().__init__(option_strings, dest, **kwargs)
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        if option_string is None:
+            raise ValueError("Missing --plugins value")
+        setattr(namespace, self.dest, option_string.split(","))
+
+
+def main():
+    logging.basicConfig(level=logging.INFO)
+    drakconfig = load_config()
+    parser = argparse.ArgumentParser(
+        description="Analyze file in Drakvuf",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument("sample_path", help="Path to the sample")
+    parser.add_argument(
+        "--vm-id",
+        help="Virtual machine ID to use for analysis",
+        type=int,
+        required=True,
+    )
+    parser.add_argument(
+        "--output-dir",
+        help="Path where analysis results will be written",
+        required=True,
+    )
+    parser.add_argument(
+        "--plugins",
+        help="DRAKVUF plugins to enable during analysis",
+        default=drakconfig.drakvuf_plugins.get_plugin_list(),
+        action=PluginsArgAction,
+    )
+    parser.add_argument(
+        "--timeout",
+        default=drakconfig.drakrun.analysis_timeout,
+        type=int,
+        help="Timeout in seconds for analysis",
+    )
+    parser.add_argument(
+        "--hooks-path", help="Path to the custom hook list", required=False
+    )
+    parser.add_argument(
+        "--start-command", help="Alternative command to start analysis", required=False
+    )
+    parser.add_argument(
+        "--extension",
+        help="Alternative extension indicating sample format",
+        required=False,
+    )
+    parser.add_argument(
+        "--sample-filename", help="Target file name for sample", required=False
```

Comment:
```suggestion
        "--sample-filename", help="Sample filename to use in the VM", required=False
```
Assuming I understand the goal correctly.